batch.py

```diff
@@ -240,6 +240,14 @@ def prepare_input_files(
     to_path(genotype_file_bed).copy('temp.bed')  # bed
     to_path(genotype_file_bim).copy('temp.bim')  # bim
     to_path(genotype_file_fam).copy('temp.fam')  # fam
+
+    # the bim file records info on variants, check that it is not empty
+    with open('temp.bim') as f:
+        for line in f.readlines():
+            if not line.strip():
+                logging.info(f'No variants found for {gene_name}')
+                return None
```

Comment:
This will check if `any` line is empty, if you're guaranteed the first line isn't empty, then you could just check the first line:

```suggestion
    with open('temp.bim') as f:
        if not f.readline().strip():
            logging.info(f'No variants found for {gene_name}')
            return None
```

Reply:
I assume the file is completely empty, so this should work!